src/lib/structures/game/Game.ts

```diff
@@ -59,34 +67,25 @@ export class Game {
 	public readonly hostUser: User;
 	public readonly guild: Guild;
 	public readonly acceptedAnswer: AcceptedAnswer;
-
+	public readonly gameType: GameType;
+	public readonly roundData: RoundData;
 	/**
 	 * The number of points to play to. Optionally provided by the user
 	 * per-game.
 	 */
 	public readonly goal?: number;
 	public players: Map<Snowflake, Player>;
 
-	/**
-	 * When playing with `AcceptedAnswer.Both`, this property is used to store
-	 * which has been guessed this round, if any.
-	 */
-	public guessedThisRound?: AcceptedAnswer.Song | AcceptedAnswer.Artist;
-
-	/**
-	 * The person(s) who guessed correctly this round, if any. Will be multiple
-	 * for `AcceptedAnswer.Both`.
-	 */
-	public guessersThisRound: User[] = [];
-
 	private readonly startTime = Date.now();
 
-	public constructor(data: GameData) {
+	public constructor(data: GameData, gameType: GameType) {
```

Comment:
Yes thank you, I wasn't sure how best to deal with this. This seems slightly better but then you have to define a new is*Game function in Game for every gametype, I suppose there's not going to be many though.

Reply:
> I think the best way to use `GameType` is with Typeguards and instead of constructor params, just inline typing and assigning in the target class. https://tsplay.dev/NBj1gN

Why would you not just test if someGame.gameType === GameType.Standard ? What is the need for the isStandardGame() method?

Reply:
The `Game` base class isn't aware of any of the classes that extend it, so if you do: 
```ts
declare const game: Game;
if (game.type === GameType.Standard) {
  // game is still typed `Game`
}
```
What you'd *have* to do is:
```ts
declare const game: StandardGame | OtherGame;
if (game.type === GameType.Standard) {
  // game is typed `StandardGame`
}
```

But that's double (or triple, quad., etc., if we make more modes) the type length and imports.

We could make a union type of:
```ts
type UnknownGame = StandardGame | OtherGame;
```
And always use that if it seems like a better option to you?

Reply:
My thought was that the Game type should be sufficient to use everywhere. Every game mode should have a guess() method, start() method, onTrackEnd() method etc... What scenario would you need to have the game typed as its subclass?

Reply:
That's fair